tools/target-server-validator/README.md

```diff
@@ -0,0 +1,90 @@
+# Apigee Target Server Validator
+
+The objective of this tool to validate targets in Target Servers & Apigee API Proxy Bundles exported from Apigee OPDK/X/Hybrid.
+Validation is done by deploying a sample proxy which check if HOST & PORT is open from Apigee OPDK/X/Hybrid.
+
+> **NOTE**: Discovery of Targets in API Proxy & Sharedflows is limited to only parsing URL from `TargetEndpoint` & `ServiceCallout` Policy.
+
+> **NOTE**: Dynamic targets are **NOT** supported, Ex : `https://host.{request.formparam.region}.example.com}`
+
+## Disclaimer
+This is not an Officially Supported Google Product!
+
+## Pre-Requisites
+* python3.x
+* Please Install required Python Libs 
+
+```
+    python3 -m pip install requirements.txt
+```
+* Please fill in `input.properties`
+
+```
+[source]
+baseurl=http://34.131.144.184:8080/v1       # Apigee OPDK/Edge/X/Hybrid Base URL
+org=xxx-xxxx-xxx-xxxxx                      # Apigee OPDK/Edge/X/Hybrid Org
+auth_type=basic                             # API Auth type basic | oauth
```

Comment:
Beware that `auth_type=oauth` does not necessarily mean that it's an X/hybrid org, because OAuth can be enabled and used to access Management APIs in Edge: https://docs.apigee.com/api-platform/system-administration/using-oauth2